src/profile/v100/app_preferences/asset_preferences/asset_preferences.rs

```diff
@@ -29,17 +29,17 @@ impl AssetPreferences {
     }
 
     pub fn hide_asset(&mut self, asset: AssetAddress) {
-        if !self.update_with(asset.id(), |x| {
-            x.set_visibility(AssetVisibility::Hidden)
-        }) {
+        if !self
+            .update_with(asset.id(), |x| x.visibility = AssetVisibility::Hidden)
```

Comment:
I **THINK** you can use write `update_with(asset, ..` ?

Reply:
Update with takes an impl Borrow. I thought it auto borrowed, but im probably mistaken